pkg/utils/gardener/cloudprofile_test.go

```diff
@@ -0,0 +1,139 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package gardener_test
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.uber.org/mock/gomock"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	pkgclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	gardenerutils "github.com/gardener/gardener/pkg/utils/gardener"
+	mockclient "github.com/gardener/gardener/third_party/mock/controller-runtime/client"
+)
+
+var _ = Describe("CloudProfile", func() {
+	Describe("#GetCloudProfile", func() {
+
+		var (
+			ctrl *gomock.Controller
+			c    *mockclient.MockClient
+
+			ctx     = context.TODO()
+			fakeErr = fmt.Errorf("fake err")
+
+			namespaceName              = "foo"
+			cloudProfileName           = "profile-1"
+			namespacedCloudProfileName = "n-profile-1"
```

Comment:
Can you initialize the values in `BeforeEach` to be safe from test modification?